lib/pwnlib/constants/constants.rb

```diff
@@ -36,17 +37,23 @@ def respond_to_missing?(method, _include_all)
       # @return [Constant]
       #   The evaluated result.
       #
+      # @raise [Pwnlib::Errors::ConstantNotFoundError]
+      #   If no value provided for variables.
+      #
       # @example
       #   eval('O_CREAT')
       #   #=> Constant('(O_CREAT)', 0x40)
       #   eval('O_CREAT | O_APPEND')
       #   #=> Constant('(O_CREAT | O_APPEND)', 0x440)
+      # @example
+      #   eval('meow')
+      #   # Pwnlib::Errors::ConstantNotFoundError: no value provided for variables: meow
       def eval(str)
         return str unless str.instance_of?(String)
         begin
           val = calculator.evaluate!(str.strip).to_i
         rescue Dentaku::UnboundVariableError => e
-          raise NameError, e.message
+          raise ::Pwnlib::Errors::ConstantNotFoundError, e.message
```

Comment:
use `e. unbound_variables` to get undefined variables, and use a more meaningful description
`"Constants are not defined: #{e.unbound_variables.join(', ')}"`